pkg/datarepair/queue/queue.go

```diff
@@ -0,0 +1,82 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package queue
+
+import (
+	"encoding/binary"
+	"sync"
+	"time"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/zeebo/errs"
+
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/storage"
+	"storj.io/storj/storage/redis"
+)
+
+//RepairQueue is the interface for the data repair queue
+type RepairQueue interface {
+	Add(qi *pb.InjuredSegment) (storage.Key, error)
+	Remove(date storage.Key) error
+	GetNext() (storage.Key, pb.InjuredSegment, error)
+	GetSize() (int, error)
+}
+
+//Queue implements the RepairQueue interface
+type Queue struct {
+	DB    storage.KeyValueStore
+	mutex *sync.Mutex
+}
+
+var (
+	queueError = errs.Class("data repair queue error")
+)
+
+//NewQueue returns a pointer to a new Queue instance with an initialized connection to Redis
+func NewQueue(address, password string, db int) (*Queue, error) {
+	rc, err := redis.NewClient(address, password, db)
```

Comment:
Pass in the database implementation rather than instantiating it here. That way it can be tested without using redis and also used without redis.